backend/open_webui/main.py

```
@@ -778,128 +1567,445 @@ async def inspect_websocket(request: Request, call_next):
+                        default_models[0].strip() if default_models[0] else None
+                    )
+
+                    if (
+                        fallback_model_id
+                        and fallback_model_id in request.app.state.MODELS
+                    ):
+                        # Update model and form_data so routing uses the fallback model's type
+                        model = request.app.state.MODELS[fallback_model_id]
+                        form_data["model"] = fallback_model_id
+                    else:
+                        raise Exception("Model not found")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)